### ISSUE TITLE
DynamicResource not working for inner views

### ISSUE THREAD

**Issue by MuneeshKumarG** [dotnet/maui#3745]
### Description

We have created a simple custom control that has multiple inner views. DynamicResource only works for the main custom control view and does not work for inner views.

Find the code for simple custom control implementation. We have implemented IVisualTreeElement also. 

```
 public class ViewExt : View, IContentView, IVisualTreeElement
    {
        public Grid ParentGrid { get; set; }
        public ViewExt()
        {
            ParentGrid = new Grid();
        }
        public View BackgroundContent
        {
            get { return (View)this.GetValue(BackgroundContentProperty); }
            set { this.SetValue(BackgroundContentProperty, value); }
        }

        public static readonly BindableProperty BackgroundContentProperty =
           BindableProperty.Create(nameof(BackgroundContent), typeof(View), typeof(ViewExt), null, propertyChanged: OnBackgroundContentPropertyChanged);

        private static void OnBackgroundContentPropertyChanged(BindableObject bindable, object oldValue, object newValue)
        {
            if(newValue is View view && bindable is ViewExt view1)
            {
                view1.ParentGrid.Children.Add(view);
            }
        }

        IReadOnlyList<IVisualTreeElement> IVisualTreeElement.GetVisualChildren()
        {
            List<Element> _logicalChildren = new();
           
            if (BackgroundContent != null)
                _logicalChildren.Add(BackgroundContent);

            return _logicalChildren.AsReadOnly();
        }

        object IContentView.Content
        {
            get
            {
                return ParentGrid;
            }
        }

        IView? IContentView.PresentedContent
        {
            get
            {
                return ParentGrid;
            }
        }

        Thickness IPadding.Padding
        {
            get
            {
                return new Thickness(0);
            }
        }

        Size IContentView.CrossPlatformMeasure(double widthConstraint, double heightConstraint)
        {
            return this.MeasureContent(widthConstraint, heightConstraint);
        }

        Size IContentView.CrossPlatformArrange(Rectangle bounds)
        {
            this.ArrangeContent(bounds);
            return bounds.Size;
        }
    }
```

```
 <VerticalStackLayout x:Name="layout">
        <VerticalStackLayout.Resources>
            <Grid Background="yellow" x:Key="bg">
                <Label Text="Background" HorizontalOptions="Center"/>
            </Grid>
            <Grid Background="red" x:Key="bg1">
                <Label Text="Background" HorizontalOptions="Center"/>
            </Grid>
        </VerticalStackLayout.Resources>

     <!--Not working-->
        <local:ViewExt >
            <local:ViewExt.BackgroundContent>
                <local:ViewExt BackgroundContent="{DynamicResource bg1}"/>
            </local:ViewExt.BackgroundContent>
        </local:ViewExt>
        
        <!--Working-->
        <local:ViewExt BackgroundContent="{DynamicResource bg}"/>

</VerticalStackLayout>

```

### Steps to Reproduce
[DynamicResourceSample.zip](https://github.com/dotnet/maui/files/7703201/DynamicResourceSample.zip)

1. Run the sample. 
2. Set the break point in C# OnBackgroundContentPropertyChanged method. 
3. See the break point not hit for inner view BackgroundContent change. 
4. Breakpoint hit only for outer view property change alone. 

### Version with bug

Preview 10 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 10

### Did you find any workaround?

**StaticResource working**

 <local:ViewExt >
            <local:ViewExt.BackgroundContent>
                <local:ViewExt BackgroundContent="{StaticResource bg1}"/>
            </local:ViewExt.BackgroundContent>
        </local:ViewExt>

**DynamicResource not working**

<local:ViewExt >
            <local:ViewExt.BackgroundContent>
                <local:ViewExt BackgroundContent="{DynamicResource bg1}"/>
            </local:ViewExt.BackgroundContent>
        </local:ViewExt>

**Comment by dotMorten** [comment]
Just confirming I'm seeing this with Preview 11 as well

**Comment by kristinx0211** [comment]
Repro on Version 17.2.0 Preview 2.0 [32216.282.main].

**Comment by StephaneDelcroix** [comment]
I'm wondering if Bindings are working in this situation...

**Comment by VincentBu** [comment]
Repro with vs main(32507.29.main).
```
<local:ViewExt >
            <local:ViewExt.BackgroundContent>
                <local:ViewExt BackgroundContent="{DynamicResource bg1}"/>
            </local:ViewExt.BackgroundContent>
</local:ViewExt>
```
we expect to have red background like this
![image](https://user-images.githubusercontent.com/44959937/167333741-014e3c0c-2f44-4d00-a4db-6a1be7b3d7ff.png)

but the background isn't rendered
![image](https://user-images.githubusercontent.com/44959937/167333842-a02ee601-f792-4fb0-b6a0-46b48d985aec.png)

**Comment by StephaneDelcroix** [comment]
it looks like ParentGrid is never Parented, probably never gets BindingContext applied, and can't walk the tree for DynamicResources propagations

**Comment by QianaJiao** [comment]
Verified this issue still repro on Visual Studio Enterprise 17.6.0 Preview 2.0  with sample project.

**Comment by gabsamples6** [comment]
This is still an issue in latest visual studio .net 7 and prevents us switching from darkMode to light mode on some controls . Most of the syncfusion controls do not work in dark mode when using DynamicResource . We do not use appthemebinding but we have multiple resource dictionaries .. LightMode -DarkMode etc.. and we swap them at runtime so DynamicResource is critical for this to work... 

Any indications if this will be looked at all? Many thanks

**Comment by Ahamed-Ali** [comment]
Any update on this?

**Comment by samhouts** [comment]
I believe that this will be fixed in NET 8 if you use https://github.com/dotnet/maui/pull/16046. It should set the Parent and the BindingContext if you use AddLogicalChild. Are you able to test with 8.0 preview 7?

**Comment by MattePozzy** [comment]
> I believe that this will be fixed in NET 8 if you use #16046. It should set the Parent and the BindingContext if you use AddLogicalChild. Are you able to test with 8.0 preview 7?

Still an issue on net 8.0.1 and VS 17.8.4.

**Comment by Redth** [comment]
I looked at your repro...

It's still net6.0 based, so just checking, even though you've installed net8 and vs 17.8.4, is your project targeting net8.0?  The fixes/changes will be available only when targeting net8.0.

After I took your repro and put it into a net8.0 project, I could see the issue happening, however what I did was this:

1. Remove `IVisualTreeElement` interface
2. Remove the implemented members for that interface
3. Instead, use `this.AddLogicalChild(ParentGrid)` that was suggested, in your `ViewExt()` ctor (this Parent Grid was not in the logical tree, and you're adding the background content view to that element).

After this, the content shows up as expected here.
